### PR TITLE
storage: flush/sync contents to disk on file close

### PR DESCRIPTION
**Pull request by rchincha**
Behavior controlled by configuration (default=off)

Signed-off-by: Ramkumar Chinchani <rchincha@cisco.com>

**What type of PR is this?**

feature

**Which issue does this PR fix**:

1. Adds a new config param to "commit" blobs on file write/close
2. Had to clean up fault injection to increase coverage on new changes and tests can be run in parallel

**What does this PR do / Why do we need it**:

In some situations, it may be required to sync/flush contents to disk immediately after writing.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rchincha** [comment]
Another option is to tune this system-wide:
/proc/sys/vm/dirty_writeback_centisecs

**Comment by rchincha** [comment]
Pls. don't merge this yet.

**Comment by rchincha** [comment]
Pls. review this, it is ready to go.

**Review by peusebiu — APPROVED** [review]
LGTM

**Review by adodon2go — APPROVED** [review]
LGTM

File: pkg/storage/s3/storage.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (is *ObjectStorage) DirExists(d string) bool {
 
 // NewObjectStorage returns a new image store backed by cloud storages.
 // see https://github.com/docker/docker.github.io/tree/master/registry/storage-drivers
-func NewImageStore(rootDir string, gc bool, dedupe bool, log zlog.Logger, metrics monitoring.MetricServer,
+func NewImageStore(rootDir string, gc, dedupe, syncOnClose bool, log zlog.Logger, metrics monitoring.MetricServer,
```
[the remark this answers is not in the document]
Comment (rchincha) [reply pkg/storage/s3/storage.go]: Let's keep "commit". Thx for the catch.

**Review by andaaron — DISMISSED** [review]

**Review by andaaron — APPROVED** [review]

**Pull request merged**